primitives/evm/src/lib.rs

```diff
@@ -91,29 +92,12 @@ pub struct WeightInfo {
 }
 
 impl WeightInfo {
-	pub fn new_from_weight_limit(
-		weight_limit: Option<Weight>,
-		proof_size_base_cost: Option<u64>,
-	) -> Result<Option<Self>, &'static str> {
-		Ok(match (weight_limit, proof_size_base_cost) {
-			(None, _) => None,
-			(Some(weight_limit), Some(proof_size_base_cost))
-				if weight_limit.proof_size() >= proof_size_base_cost =>
-			{
-				Some(WeightInfo {
-					ref_time_limit: Some(weight_limit.ref_time()),
-					proof_size_limit: Some(weight_limit.proof_size()),
-					ref_time_usage: Some(0u64),
-					proof_size_usage: Some(proof_size_base_cost),
-				})
-			}
-			(Some(weight_limit), None) => Some(WeightInfo {
-				ref_time_limit: Some(weight_limit.ref_time()),
-				proof_size_limit: None,
-				ref_time_usage: Some(0u64),
-				proof_size_usage: None,
-			}),
-			_ => return Err("must provide Some valid weight limit or None"),
+	pub fn new_from_weight_limit(weight_limit: Option<Weight>) -> Option<Self> {
+		weight_limit.map(|weight_limit| WeightInfo {
+			ref_time_limit: Some(weight_limit.ref_time()),
+			proof_size_limit: Some(weight_limit.proof_size()),
+			ref_time_usage: Some(0u64),
+			proof_size_usage: Some(0u64),
```

Comment:
Solochains have no proof_size limitations 
```suggestion
	pub fn new_from_weight_limit(weight_limit: Option<Weight>, recording_proof_size: bool) -> Option<Self> {
		weight_limit.map(|weight_limit| WeightInfo {
			ref_time_limit: Some(weight_limit.ref_time()),
			proof_size_limit: if recording_proof_size { Some(weight_limit.proof_size()) } else { None },
			ref_time_usage: Some(0u64),
			proof_size_usage: Some(0u64),
```